Bottom: a8bebc02722556c48ac1bf19e88dafbe5d1d0284
Top:    f5a615c01b04074d0712cdeb75b04ed076df2b2c
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2011-04-20 21:41:18 +0100

Remove Erase* functions in favour of invalidating regions


---

diff --git a/src/draw.c b/src/draw.c
index 8d6e0f6..eb9528a 100644
--- a/src/draw.c
+++ b/src/draw.c
@@ -1916,187 +1916,6 @@ DrawElementPinsAndPads (ElementTypePtr Element, int unused)
   END_LOOP;
 }
 
-/* ---------------------------------------------------------------------------
- * erase a via
- */
-void
-EraseVia (PinTypePtr Via)
-{
-  DrawPinOrViaLowLevel (Via, false);
-  if (TEST_FLAG (DISPLAYNAMEFLAG, Via))
-    DrawPinOrViaNameLowLevel (Via);
-}
-
-/* ---------------------------------------------------------------------------
- * erase a ratline
- */
-void
-EraseRat (RatTypePtr Rat)
-{
-  if (TEST_FLAG(VIAFLAG, Rat))
-    {
-      int w = Rat->Thickness;
-
-      if (TEST_FLAG (THINDRAWFLAG, PCB))
-	gui->set_line_width (Output.fgGC, 0);
-      else
-	gui->set_line_width (Output.fgGC, w);
-      gui->draw_arc (Output.fgGC, Rat->Point1.X, Rat->Point1.Y,
-		     w * 2, w * 2, 0, 360);
-    }
-  else
-    DrawLineLowLevel ((LineTypePtr) Rat);
-}
-
-
-/* ---------------------------------------------------------------------------
- * erase a via name
- */
-void
-EraseViaName (PinTypePtr Via)
-{
-  DrawPinOrViaNameLowLevel (Via);
-}
-
-/* ---------------------------------------------------------------------------
- * erase a pad object
- */
-void
-ErasePad (PadTypePtr Pad)
-{
-  DrawPadLowLevel (Output.fgGC, Pad, false, false);
-  if (TEST_FLAG (DISPLAYNAMEFLAG, Pad))
-    DrawPadNameLowLevel (Pad);
-}
-
-/* ---------------------------------------------------------------------------
- * erase a pad name
- */
-void
-ErasePadName (PadTypePtr Pad)
-{
-  DrawPadNameLowLevel (Pad);
-}
-
-/* ---------------------------------------------------------------------------
- * erase a pin object
- */
-void
-ErasePin (PinTypePtr Pin)
-{
-  DrawPinOrViaLowLevel (Pin, false);
-  if (TEST_FLAG (DISPLAYNAMEFLAG, Pin))
-    DrawPinOrViaNameLowLevel (Pin);
-}
-
-/* ---------------------------------------------------------------------------
- * erase a pin name
- */
-void
-ErasePinName (PinTypePtr Pin)
-{
-  DrawPinOrViaNameLowLevel (Pin);
-}
-
-/* ---------------------------------------------------------------------------
- * erases a line on a layer
- */
-void
-EraseLine (LineTypePtr Line)
-{
-  DrawLineLowLevel (Line);
-}
-
-/* ---------------------------------------------------------------------------
- * erases an arc on a layer
- */
-void
-EraseArc (ArcTypePtr Arc)
-{
-  if (!Arc->Thickness)
-    return;
-  DrawArcLowLevel (Arc);
-}
-
-/* ---------------------------------------------------------------------------
- * erases a text on a layer
- */
-void
-EraseText (LayerTypePtr Layer, TextTypePtr Text)
-{
-  int min_silk_line;
-  if (Layer == & PCB->Data->SILKLAYER
-      || Layer == & PCB->Data->BACKSILKLAYER)
-    min_silk_line = PCB->minSlk;
-  else
-    min_silk_line = PCB->minWid;
-  DrawTextLowLevel (Text, min_silk_line);
-}
-
-/* ---------------------------------------------------------------------------
- * erases a polygon on a layer
- */
-void
-ErasePolygon (PolygonTypePtr Polygon)
-{
-  DrawPolygonLowLevel (Polygon);
-}
-
-/* ---------------------------------------------------------------------------
- * erases an element
- */
-void
-EraseElement (ElementTypePtr Element)
-{
-  ELEMENTLINE_LOOP (Element);
-  {
-    DrawLineLowLevel (line);
-  }
-  END_LOOP;
-  ARC_LOOP (Element);
-  {
-    DrawArcLowLevel (arc);
-  }
-  END_LOOP;
-  if (!TEST_FLAG (HIDENAMEFLAG, Element))
-    DrawTextLowLevel (&ELEMENT_TEXT (PCB, Element), PCB->minSlk);
-  EraseElementPinsAndPads (Element);
-}
-
-/* ---------------------------------------------------------------------------
- * erases all pins and pads of an element
- */
-void
-EraseElementPinsAndPads (ElementTypePtr Element)
-{
-  PIN_LOOP (Element);
-  {
-    DrawPinOrViaLowLevel (pin, false);
-    if (TEST_FLAG (DISPLAYNAMEFLAG, pin))
-      DrawPinOrViaNameLowLevel (pin);
-  }
-  END_LOOP;
-  PAD_LOOP (Element);
-  {
-    DrawPadLowLevel (Output.fgGC, pad, false, false);
-    if (TEST_FLAG (DISPLAYNAMEFLAG, pad))
-      DrawPadNameLowLevel (pad);
-  }
-  END_LOOP;
-}
-
-/* ---------------------------------------------------------------------------
- * erases the name of an element
- */
-void
-EraseElementName (ElementTypePtr Element)
-{
-  if (TEST_FLAG (HIDENAMEFLAG, Element))
-    return;
-  DrawTextLowLevel (&ELEMENT_TEXT (PCB, Element), PCB->minSlk);
-}
-
-
 void
 EraseObject (int type, void *lptr, void *ptr)
 {
@@ -2133,8 +1952,6 @@ EraseObject (int type, void *lptr, void *ptr)
     }
 }
 
-
-
 void
 DrawObject (int type, void *ptr1, void *ptr2, int unused)
 {
